### DISCUSSION TITLE
Can we use dynamic bandwidth?

### DISCUSSION THREAD

**Discussion by sqsssq** [d3/d3#3762] [Q&A]
when I use d3.contourDensity, I hope I can input dynamic bandwidths. If that, I can have the result of adaptive KDE.

**Comment by Fil** [comment]
You could compute the bandwidth with NRD, see the discussion at https://github.com/d3/d3-contour/issues/27 and the prototype at https://observablehq.com/@fil/x-y-bandwidth-for-density-contours
